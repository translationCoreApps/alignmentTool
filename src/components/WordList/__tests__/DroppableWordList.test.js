/* eslint-env jest */
import {Token} from 'wordmap-lexer';
import React, {Component} from 'react';
import TestBackend from 'react-dnd-test-backend';
import {DragDropContext} from 'react-dnd';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DroppableWordList from '../index';

describe('Test DroppableWordList component in WordList/index.js', () => {
  let props;

  beforeEach(() => {
    const w3 = new Token({text: 'w3'});
    w3.disabled = true;
    const words = [
      new Token({text: 'w1'}),
      new Token({text: 'w2', occurrence: 1, occurrences: 2}),
      new Token({text: 'w2', occurrence: 2, occurrences: 2}),
      w3
    ];
    props = {
      chapter: 1,
      verse: 1,
      onDropTargetToken: jest.fn(),
      words: words,
      moveBackToWordBank: jest.fn(),
      modalOpen: false,
      toolsSettings: {}
    };
  });

  test('DroppableWordList component full integration test', () => {
    const WrappedWordList = wrapInTestContext(DroppableWordList);
    const wrapper = mount(<WrappedWordList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('DroppableWordList test scroll save and restore', () => {
    const WrappedWordList = wrapInTestContext(DroppableWordList);
    props.wordList = {
      scrollTop: 100
    };
    const wrapper = mount(<WrappedWordList {...props} />);
    wrapper.setProps({isOver: true});
    wrapper.setProps({isOver: false});
    wrapper.setProps({modalOpen: false});
    wrapper.setProps({chapter: 2});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

//
// Helpers
//

/**
 * Wraps a component into a DragDropContext that uses the TestBackend.
 */
function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(
    class TestContextContainer extends Component {
      render() {
        return <DecoratedComponent {...this.props} />;
      }
    }
  );
}
