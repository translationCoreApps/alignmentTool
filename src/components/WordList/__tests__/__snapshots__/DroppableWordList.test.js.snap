// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DroppableWordList component in WordList/index.js DroppableWordList component full integration test 1`] = `
<DragDropContext(TestContextContainer)
  chapter={1}
  moveBackToWordBank={[Function]}
  onDropTargetToken={[Function]}
  verse={1}
  words={
    Array [
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 1,
      },
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 2,
      },
      Object {
        "index": 0,
        "occurrence": 2,
        "occurrences": 2,
      },
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 1,
      },
    ]
  }
>
  <TestContextContainer
    chapter={1}
    moveBackToWordBank={[Function]}
    onDropTargetToken={[Function]}
    verse={1}
    words={
      Array [
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 1,
        },
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 2,
        },
        Object {
          "index": 0,
          "occurrence": 2,
          "occurrences": 2,
        },
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 1,
        },
      ]
    }
  >
    <DropTarget(DroppableWordList)
      chapter={1}
      moveBackToWordBank={[Function]}
      onDropTargetToken={[Function]}
      verse={1}
      words={
        Array [
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 1,
          },
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 2,
          },
          Object {
            "index": 0,
            "occurrence": 2,
            "occurrences": 2,
          },
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 1,
          },
        ]
      }
    >
      <DroppableWordList
        canDrop={false}
        chapter={1}
        connectDropTarget={[Function]}
        isOver={false}
        moveBackToWordBank={[Function]}
        onDropTargetToken={[Function]}
        verse={1}
        words={
          Array [
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 1,
            },
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 2,
            },
            Object {
              "index": 0,
              "occurrence": 2,
              "occurrences": 2,
            },
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 1,
            },
          ]
        }
      >
        <div
          id="wordList"
          style={
            Object {
              "backgroundColor": "#DCDCDC",
              "flex": 0.2,
              "overflowY": "auto",
              "padding": "5px 8px 5px 5px",
              "width": "100%",
            }
          }
        >
          <WordList
            chapter={1}
            isOver={false}
            verse={1}
            words={
              Array [
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 1,
                },
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 2,
                },
                Object {
                  "index": 0,
                  "occurrence": 2,
                  "occurrences": 2,
                },
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 1,
                },
              ]
            }
          >
            <div
              key="0"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 1,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 1,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={1}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w1"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w1
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={1}
                          />
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="1"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 2,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 2,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={2}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w2"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w2
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={2}
                          >
                            <sup
                              style={
                                Object {
                                  "marginTop": "2px",
                                  "opacity": "0.8",
                                  "top": 0,
                                }
                              }
                            >
                              1
                            </sup>
                          </WordOccurrence>
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="2"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 2,
                    "occurrences": 2,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 2,
                      "occurrences": 2,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={2}
                      occurrences={2}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w2"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w2
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={2}
                            occurrences={2}
                          >
                            <sup
                              style={
                                Object {
                                  "marginTop": "2px",
                                  "opacity": "0.8",
                                  "top": 0,
                                }
                              }
                            >
                              2
                            </sup>
                          </WordOccurrence>
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="3"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={true}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 1,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={true}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 1,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={true}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={1}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w3"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #868686",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "not-allowed",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 0.3,
                              "padding": "10px",
                              "userSelect": "none",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w3
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={1}
                          />
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
          </WordList>
        </div>
      </DroppableWordList>
    </DropTarget(DroppableWordList)>
  </TestContextContainer>
</DragDropContext(TestContextContainer)>
`;

exports[`Test DroppableWordList component in WordList/index.js DroppableWordList test scroll save and restore 1`] = `
<DragDropContext(TestContextContainer)
  chapter={2}
  isOver={false}
  moveBackToWordBank={[Function]}
  onDropTargetToken={[Function]}
  verse={1}
  wordList={
    Object {
      "scrollTop": 0,
    }
  }
  words={
    Array [
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 1,
      },
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 2,
      },
      Object {
        "index": 0,
        "occurrence": 2,
        "occurrences": 2,
      },
      Object {
        "index": 0,
        "occurrence": 1,
        "occurrences": 1,
      },
    ]
  }
>
  <TestContextContainer
    chapter={2}
    isOver={false}
    moveBackToWordBank={[Function]}
    onDropTargetToken={[Function]}
    verse={1}
    wordList={
      Object {
        "scrollTop": 0,
      }
    }
    words={
      Array [
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 1,
        },
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 2,
        },
        Object {
          "index": 0,
          "occurrence": 2,
          "occurrences": 2,
        },
        Object {
          "index": 0,
          "occurrence": 1,
          "occurrences": 1,
        },
      ]
    }
  >
    <DropTarget(DroppableWordList)
      chapter={2}
      isOver={false}
      moveBackToWordBank={[Function]}
      onDropTargetToken={[Function]}
      verse={1}
      wordList={
        Object {
          "scrollTop": 0,
        }
      }
      words={
        Array [
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 1,
          },
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 2,
          },
          Object {
            "index": 0,
            "occurrence": 2,
            "occurrences": 2,
          },
          Object {
            "index": 0,
            "occurrence": 1,
            "occurrences": 1,
          },
        ]
      }
    >
      <DroppableWordList
        canDrop={false}
        chapter={2}
        connectDropTarget={[Function]}
        isOver={false}
        moveBackToWordBank={[Function]}
        onDropTargetToken={[Function]}
        verse={1}
        wordList={
          Object {
            "scrollTop": 0,
          }
        }
        words={
          Array [
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 1,
            },
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 2,
            },
            Object {
              "index": 0,
              "occurrence": 2,
              "occurrences": 2,
            },
            Object {
              "index": 0,
              "occurrence": 1,
              "occurrences": 1,
            },
          ]
        }
      >
        <div
          id="wordList"
          style={
            Object {
              "backgroundColor": "#DCDCDC",
              "flex": 0.2,
              "overflowY": "auto",
              "padding": "5px 8px 5px 5px",
              "width": "100%",
            }
          }
        >
          <WordList
            chapter={2}
            isOver={false}
            verse={1}
            words={
              Array [
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 1,
                },
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 2,
                },
                Object {
                  "index": 0,
                  "occurrence": 2,
                  "occurrences": 2,
                },
                Object {
                  "index": 0,
                  "occurrence": 1,
                  "occurrences": 1,
                },
              ]
            }
          >
            <div
              key="0"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 1,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 1,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={1}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w1"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w1
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={1}
                          />
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="1"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 2,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 2,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={2}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w2"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w2
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={2}
                          >
                            <sup
                              style={
                                Object {
                                  "marginTop": "2px",
                                  "opacity": "0.8",
                                  "top": 0,
                                }
                              }
                            >
                              1
                            </sup>
                          </WordOccurrence>
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="2"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={false}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 2,
                    "occurrences": 2,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={false}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 2,
                      "occurrences": 2,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={false}
                      isSuggestion={false}
                      occurrence={2}
                      occurrences={2}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w2"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #44C6FF",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "pointer",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 1,
                              "padding": "10px",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w2
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={2}
                            occurrences={2}
                          >
                            <sup
                              style={
                                Object {
                                  "marginTop": "2px",
                                  "opacity": "0.8",
                                  "top": 0,
                                }
                              }
                            >
                              2
                            </sup>
                          </WordOccurrence>
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
            <div
              key="3"
              style={
                Object {
                  "margin": "10px",
                }
              }
            >
              <DragSource(SecondaryToken)
                disabled={true}
                token={
                  Object {
                    "index": 0,
                    "occurrence": 1,
                    "occurrences": 1,
                  }
                }
              >
                <SecondaryToken
                  connectDragSource={[Function]}
                  disabled={true}
                  isDragging={false}
                  token={
                    Object {
                      "index": 0,
                      "occurrence": 1,
                      "occurrences": 1,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <WordCard
                      disabled={true}
                      isSuggestion={false}
                      occurrence={1}
                      occurrences={1}
                      onCancel={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      word="w3"
                    >
                      <div
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "borderLeft": "5px solid #868686",
                              "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                              "cursor": "not-allowed",
                              "display": "flex",
                              "flexDirection": "row",
                              "opacity": 0.3,
                              "padding": "10px",
                              "userSelect": "none",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                              }
                            }
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "flexGrow": 2,
                                }
                              }
                            >
                              w3
                            </span>
                          </span>
                          <WordOccurrence
                            occurrence={1}
                            occurrences={1}
                          />
                        </div>
                      </div>
                    </WordCard>
                  </div>
                </SecondaryToken>
              </DragSource(SecondaryToken)>
            </div>
          </WordList>
        </div>
      </DroppableWordList>
    </DropTarget(DroppableWordList)>
  </TestContextContainer>
</DragDropContext(TestContextContainer)>
`;
